models/Redirect/Dao.php

```diff
@@ -68,7 +68,7 @@ public function getByExactMatch(Request $request, ?Site $site = null, bool $over
             ) AND active = 1 AND regex IS NULL AND (expiry > UNIX_TIMESTAMP() OR expiry IS NULL)';
 
         if ($siteId) {
-            $sql .= ' AND sourceSite = ' . $this->db->quote($siteId);
+            $sql .= ' AND (sourceSite IS NULL OR sourceSite = ' . $this->db->quote($siteId).')';
         } else {
             $sql .= ' AND sourceSite IS NULL';
         }
```

Comment:
Thank you! 
Leaving some notes as a confirmation is needed.

If https://github.com/pimcore/pimcore/pull/14954#issuecomment-1594713300 is valid, we can remove the `if ($siteId)` and else part. 
- When `siteId` is `0`, it should be checking `sourceSite IS NULL OR 0`
- we also need a migration to change all former `NULL` (which before PR targeted mainSite only) to be `0`
- Upgrade notes for these changes would be necessary

Reply:
@BlackbitDevs please see ☝️ Also, we need another option in the dropdown (e.g. Empty) to make the site optional again once it is saved.

Reply:
@BlackbitDevs Sorry for late reply 😬 
I agree with @kingjia90 proposed solution - it's safe in terms of BC-breaks, but it's worth to mention in upgrade notes, because you now have more possibilities. 
Can you maybe update your PR accordingly? 
Thanks a lot in advance! 💟